index.js

```diff
@@ -4,26 +4,40 @@ var vt = require('vector-tile');
 var request = require('request');
 var Protobuf = require('pbf');
 var format = require('util').format;
+var fs = require('fs');
 
 module.exports = function(args, callback) {
-    var zxy = args.url.match(/\/(\d+)\/(\d+)\/(\d+)/);
 
-    args.x = args.x === undefined ? zxy[2] : args.x;
-    args.y = args.y === undefined ? zxy[3] : args.y;
-    args.z = args.z === undefined ? zxy[1] : args.z;
-
-    if (args.x === undefined || args.y === undefined || args.z === undefined) {
-        return callback(new Error(format("Could not determine tile z, x, and y from %s; specify manually with -z <z> -x <x> -y <y>", JSON.stringify(url))));
+    try {
+        var file = fs.lstatSync(args.uri);
+        if (file.isFile()) {
+            fs.readFile(args.uri, function(err, data) {
+                if (err) throw err;
+                readTile(data);
+            });
+        }
+    } catch(err) {
+        request.get({
+            url: args.uri,
+            gzip: true,
+            encoding: null
+        }, function (err, response, body) {
+            if (err) throw err;
+            readTile(body);
+        });
     }
```

Comment:
Another option to consider:

```js
var url = require('url')
...
var parsed = url.parse(args.uri);
if (parsed.protocol && parsed.protocol.indexOf('http') > -1) {
   ... then treat as url
} else {
   .. then treat as file
}
```

This would support `file://./fixtures/9-150-194.pbf` as well as `./fixtures/9-150-194.pbf` assuming you opened `parsed.pathname`.